### ISSUE TITLE
volta install doesn't respect pinned engines

### ISSUE THREAD

**Issue by olivierlacan** [volta-cli/volta#986]
Given: 

```
// package.json
{
  "volta": {
    "node": "14.15.0",
    "npm": "7.0.7"
  }
}
``` 

I would expect this command to install Node 14.15.0. Instead if installs 14.16.0 (currently). 

```
$ volta install node
  Fetching node@14.16.1  [========>                               ]  21%
```

I believe that a pinned engine will lead to an error if invoked with a different version than the one that's pinned, but couldn't Volta read these pinned version and enforce them to install the appropriate version as well?

**Comment by charlespierce** [comment]
Hi @olivierlacan, I think this is actually a miss in our documentation / command names. Volta uses a mental model that is different from many other Node version management tools, in that you don't have to manually or explicitly "install" or "use" a Node version. When you run `node`, Volta will detect the pinned version and automatically make it available (fetching it from nodejs.org if necessary), then run whatever Node command you specified.

The `volta install node` command fetches the specified version (if none is specified, the default is the latest LTS) and sets that as your global _default_. The default is used whenever you call `node` from a directory that is _not_ part of a project that has a pinned Node version.

In your example, you are in a project that has a pinned version, so while `volta install node` will set the default to the latest LTS, running `node -v` will actually detect and fetch Node 14.15.0 and then execute it, which will show `v14.15.0`. So you don't even need to run `volta install node` or any other command, you only need to navigate to the project that has the above `package.json` and run Node. Volta will make sure the appropriate version is executed.

Additionally, if you _do_ run `volta install node` like this case, at the end of the process, it should print out 2 messages: One saying that it set `14.16.1` as the default and another that's a note letting you know that you are using `14.15.0` in the current project, despite the fact that you just set the default.

**Comment by olivierlacan** [comment]
That's super helpful @charlespierce. So in any setup script I might have all I have to do is install Volta, ensure that any config file is reloaded, and never have to do any manual install process since any future invocation of the node binary will result in Volta installing the pinned version. 

Confirmed that the default setting occurs. I would honestly recommend a simple message signaling what Volta is doing *before* the download begins, to avoid the potential surprise. 

Example: 

```
$ volta install node
Latest Node LTS version resolved to 14.16.1
Installing and setting it as a global default...
  Fetching node@14.16.1  [========>                               ]  21%

...
success: installed and set node@14.16.1 (with npm@6.14.12) as default
   note: you are using node@14.15.0 in the current project
```

**Comment by charlespierce** [comment]
> So in any setup script I might have all I have to do is install Volta, ensure that any config file is reloaded, and never have to do any manual install process since any future invocation of the node binary will result in Volta installing the pinned version.

Yep, exactly!

> I would honestly recommend a simple message signaling what Volta is doing _before_ the download begins, to avoid the potential surprise.

Yeah, something like that is a good idea. In fact, since we have the info available up-front, we may even be able to show the `note` text (slightly edited) up-front so it's less confusing.
